### DISCUSSION TITLE
Import 3D Gltf or Obj models to the scene [Evercam]

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#2696] [Q&A] [answered]
Is there a way to import and place a 3D static model into the scene, this could be a GLTF format which is a standard web format for handeling 3D models, or an Obj.

Cesium already supports such features here is an example bellow
![image](https://user-images.githubusercontent.com/29980658/141460009-74e2d47e-1162-4719-bac9-48a3bbf97c7a.png)

**Comment by pmconne** [comment]
iTwin.js includes a glTF reader that it uses internally to read the glTF contents of 3d tiles (b3dm, i3dm, etc). But it doesn't currently expose an API for reading glTF directly.
I suspect it wouldn't take much effort to add such an API. I'll take a look.

**Reply by salah-walid** [reply]
Is there news on adding this feature ?

**Reply by pmconne** [reply]
It looks doable. Would you like to supply the model in your screenshot and/or other glTF models of interest that we can test against?

**Reply by salah-walid** [reply]
Yes here you go:
https://drive.google.com/file/d/1FY5QpA25Cct1qH01ozcf7r2oVAoAnQ0-/view?usp=sharing

**Comment by pmconne** [comment]
@salah-walid this is in progress as #2911.
![image](https://user-images.githubusercontent.com/22944042/146428878-3025d614-2f6d-423d-86b9-27c321ac2157.png)

**Reply by salah-walid** [reply]
Is this going to be released with v3 ?

**Reply by pmconne** [reply]
Yes, it will be included in 3.0.

**Comment by pmconne** [comment, marked as answer]
@salah-walid I'm marking this as answered. 3.0.0 has been released. Here's some [example code](https://www.itwinjs.org/learning/frontend/viewdecorations/#gltf-decorations) that loads a glb asset from disk. Here's [a similar example](https://github.com/iTwin/itwinjs-core/blob/master/test-apps/display-test-app/src/frontend/GltfDecoration.ts) specifying a URL to the glTF asset. Let us know if you have any further questions or issues.
